types/modules/currency-db.d.ts

```diff
@@ -0,0 +1,60 @@
+import { FirebotUser } from "./user-db";
+
+type Currency = {
+    id: string;
+    name: string;
+    active: boolean;
+    limit: number;
+    transfer: "Allow" | "Disallow";
+    interval: number;
+    payout: number;
+    /** Offline payout */
+    offline: number;
+    bonus: Record<string, number>;
+  };
+
+export type CurrencyDB = {
+    adjustCurrencyForUser: (
+        username: string, 
+        currencyId: string,
+        value: number,
+        adjustType?: string
+    ) => boolean,
+    addCurrencyToOnlineUsers: (
+        currencyId: string,
+        value: string,
+        ignoreDisposable?: boolean,
+        adjustType?: string
+    ) => Promise<void>,
+    getUserCurrencyAmount: (
+        username: string,
+        currencyId: string
+    ) => number,
+    purgeCurrencyById: (
+        currencyId: string
+    ) => void,
+    addCurrencyToNewUser: (
+        user: FirebotUser
+    ) => FirebotUser,
+    refreshCurrencyCache: () => void,
+    getCurrencies: () => Array<Currency>,
+    getCurrencyById: (
+        currencyId: string
+    ) => Currency,
+    addCurrencyToUserGroupOnlineUsers: (
```

Comment:
```suggestion
    getCurrencyByName(currencyName: string) => Currency | undefined,
    addCurrencyToUserGroupOnlineUsers: (
```
This recently introduced `getCurrencyByName` function is missing.